daemon/openrazer_daemon/misc/battery_notifier.py

```diff
@@ -73,27 +74,40 @@ def notify_battery(self):
             self._last_notify_time = now
 
             battery_level = self._get_battery_func()
+            battery_percent = int(round(battery_level, 0))
 
             # Sometimes on wifi don't get batt
             if battery_level == -1.0:
                 time.sleep(0.2)
                 battery_level = self._get_battery_func()
 
-            if battery_level < 10.0:
-                if battery_level == 0.0:
-                    # Do nothing
-                    pass
-                else:
-                    if self._notify2:
-                        self._notification.update(summary="{0} Battery at {1:.1f}%".format(self._device_name, battery_level), message='Please charge your device', icon='notification-battery-low')
-                        self._notification.show()
-            else:
-                if self._notify2:
-                    self._notification.update(summary="{0} Battery at {1:.1f}%".format(self._device_name, battery_level))
-                    self._notification.show()
+            title = self._device_name
+            message = "Battery is {0}%".format(battery_percent)
+            icon = "battery-full"
+
+            if battery_level == 0.0:
+                # Do nothing
+                pass
+
+            elif battery_level <= 10.0:
+                message = "Battery is low ({0}%). Please charge your device".format(battery_percent)
+                icon = "battery-empty"
+
+            elif battery_level <= 30.0:
+                icon = "battery-low"
+
+            elif battery_level <= 70.0:
+                icon = "battery-good"
+
+            elif battery_level == 100.0:
+                message = "Battery is fully charged ({0}%)".format(battery_percent)
```

Comment:
Are you sure the devices stay consistently at 100%? Phones generally say they're fully charged with anything >90% or >95% because of how charging works.

Reply:
I'm not sure, I have no battery powered Razer devices to know the expected behaviour when it's finished charging (raw value, `is_charging`, etc)

At least on my phone, the OS only reports "Fully charged" when it reaches 100%. So, I have no idea if the Razer firmware does anything special like that or just outputs a literal raw value that may never reach 100%. 🤷🏼‍♂️

Reply:
Got one confirmation so far: https://github.com/openrazer/openrazer/issues/1662#issuecomment-997033361. At least for the Viper Ultimate, it is confirmed to exactly 100% before the firmware stops its built-in charging effect.